types/preview/@ember/array/-private/native-array.d.ts

```diff
@@ -4,10 +4,11 @@ declare module '@ember/array/-private/native-array' {
   import MutableArray from '@ember/array/mutable';
   import Mixin from '@ember/object/mixin';
 
-  // Get an alias to the global Array type to use in inner scope below.
+  // Get aliases to the global Array types to use in inner scope below.
   type Array<T> = T[];
+  type ReadonlyArray<T> = readonly T[];
 
-  type AnyArray<T> = EmberArray<T> | Array<T>;
+  type AnyArray<T> = EmberArray<T> | ReadonlyArray<T>;
```

Comment:
So this will "just work", but we also want to keep around the ability to work explicitly with regular arrays:

```suggestion
type AnyArray<T> = EmberArray<T> | Array<T> | ReadonlyArray<T>;
```

Reply:
Doesn't `ReadonlyArray<T>` include all `Array<T>`?

Reply:
That is, an `Array<T>` can always be cast to/assigned to a `ReadonlyArray<T>` I believe?

Reply:
*Cast to*, yes, but *assigned to*, no: it depends on what position it's in in terms of variance whether it's legal to substitute one for the other. Explicitly specifying that we accept both any place we have `AnyArray<T>` works around that.

Reply:
OK, I did what you suggest. But if you have the time, can you show me an example (in the TypeScript playground for example) of when an `Array<T>` can't be passed as a `ReadonlyArray<T>`?

Reply:
Yep, any callback function parameter will get you. [example](https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEAzArgOzAFwJYHtXxAA8oBbABwhAElUAeAFQD4AKMAIwC55nYYoBPLgCUQUYHgj8AgjD78GjAJTwAvI3gy5CxVwBuOLMADcAKFCRYCFOmx4CxcpQDyyDAtaduvAV00DtquoiYhLSsv5MOvD6hqYmJtaYuPjiyGyUAKIkzKgAzlyoyCRsIDAA2gC6UYXFpZXwAN4m8C3wcBjIMPh5AHQkUGQ5gfD4AFTwAEyKJgC+8USkFNSozKnpIFmKpguOIC4YqzhpmSRb8UA)